cpp/webcrypto/crypto_ec.cpp

```diff
@@ -0,0 +1,129 @@
+//
+//  crypto_ec.cpp
+//  BEMCheckBox
+//
+//  Created by Oscar Franco on 30/11/23.
+//
+
+#include "crypto_ec.h"
+#include <openssl/ec.h>
+
+namespace margelo {
+namespace jsi = facebook::jsi;
+
+ECPointPointer ECDH::BufferToPoint(jsi::Runtime &rt,
+                                   const EC_GROUP* group,
+                                   jsi::ArrayBuffer &buf) {
+    int r;
+
+    ECPointPointer pub(EC_POINT_new(group));
+    if (!pub) {
+        throw std::runtime_error(
+            "Failed to allocate EC_POINT for a public key");
+        return pub;
+    }
+
+    // TODO(osp) re-insert this check
+    //  if (UNLIKELY(!input.CheckSizeInt32())) {
+    //    THROW_ERR_OUT_OF_RANGE(env, "buffer is too big");
+    //    return ECPointPointer();
+    //  }
+    r = EC_POINT_oct2point(
+                           group,
+                           pub.get(),
+                           buf.data(rt),
+                           buf.size(rt),
+                           nullptr);
+
+    if (!r)
+        return ECPointPointer();
+
+    return pub;
+}
+
+void PKEY_SPKI_Export(
+                      KeyObjectData* key_data,
+                      ByteSource* out) {
```

Comment:
I'm curious... is this pattern of a void function and passing in the return value as an argument the one we should use?

Or should the function return the value? 

I'm asking, because I am attempting to use `std::variant` in another PR, and am having to deal with these `ByteSource` values in `MGLWebCrypto.cpp`

Reply:
This is c style of pointer passing

Reply:
right, but is there any performance advantage?  Should we use the C++ return values style?  Should we be consistent?

Reply:
It's just the code from Node. Don't deviate too much as sometimes multiple params need to be returned and pointer passing is the easiest way. Otherwise, you will end up creating a bunch of unnecessary structs. The node code also serves as a reference for a proper implementation.